### DISCUSSION TITLE
What does the translation data structure look like in DB?

### DISCUSSION THREAD

**Discussion by momijizukamori** [spatie/laravel-translatable#440] [Q&A]
I have a project I'd like to use laravel-translatable in, but we generate our in a different system (Weblate), which means I have data files that need to be sideloaded into the database. I'm fine writing the loader myself, but I need to know what structure laravel-translatable is expecting. We have a fairly large number of entries that need to be added, so using `setTranslation` isn't really a workable option.

**Comment by youseffatouraee** [comment]
It supports 'JSON' and 'text' data types:
https://spatie.be/docs/laravel-translatable/v6/installation-setup#content-making-a-model-translatable
It also worked for me on the 'longtext' type in MySQL.


It creates records like:
`
{"en":"English Translation","fr":"French Translation"}   
`

Besides, you can use some hacks like #259 to convert your text to this type, after setting $translatable=[] property on your models.
